### PR TITLE
deps: Update github.com/aws/aws-sdk-go/...

### PR DESCRIPTION
**Pull request by jen20**
This pull request covers updating the AWS SDK to the latest revision, in order to pull in STS packages required for feature development. Submitted as a separate PR ahead of the rest of the work such that we get some acceptance test runs in with the updated SDK.

### CODE REVIEW

File: vendor/github.com/aws/aws-sdk-go/aws/config.go [hunk 1]
```diff
@@ -130,11 +130,12 @@ type Config struct {
 	// client to create a new http.Client. This options is only meaningful if you're not
 	// already using a custom HTTP client with the SDK. Enabled by default.
 	//
-	// Must be set and provided to the session.New() in order to disable the EC2Metadata
-	// overriding the timeout for default credentials chain.
+	// Must be set and provided to the session.NewSession() in order to disable
+	// the EC2Metadata overriding the timeout for default credentials chain.
```
Comment (catsby) [inline vendor/github.com/aws/aws-sdk-go/aws/config.go]: There are a handful of spots where we call `session.New`, which this PR brings in as deprecated. 

```console
$ [builtin/providers/aws][pr-8056](2)$ ack "session.New" .
auth_helpers.go
28:     metadataClient := ec2metadata.New(session.New(cfg))
117:    metadataClient := ec2metadata.New(session.New(cfg))

auth_helpers_test.go
651:    sess := session.New(&aws.Config{

config.go
181:        sess := session.New(awsConfig)
```

If we're pulling this in to get some Acc. tests ran with the new SDK, I'd like to see those changed as well, for the same reason

Comment (jen20) [inline vendor/github.com/aws/aws-sdk-go/aws/config.go]: Seems reasonable - I'll go through and do this.

**Comment by jen20** [comment]
OK this should be fixed up now @catsby!

**Comment by catsby** [comment]
There are merge conflicts now, it seems, but _besides_ those, this looks good 👍 , thanks for doing that extra bit!

**Comment by jen20** [comment]
OK I've fixed this up and merged on master.

**Pull request merged**